oauth2client/contrib/flask_util.py

```diff
@@ -443,7 +443,12 @@ def credentials(self):
 
     def has_credentials(self):
         """Returns True if there are valid credentials for the current user."""
-        return self.credentials and not self.credentials.invalid
+        if not self.credentials:
+            return False
```

Comment:
How bad does this look as one boolean expression? Is it just
```
return (
    self.credentials and
    (not self.credentials.access_token_expired or self.credentials.refresh_token))
```
?

If that's too unattractive, please change
```
if <condition>:
  return False
if <other condition>:
  return False
return True
```
to
```
if <condition>:
  return False
elif <other condition>:
  return False
else:
  return True
```
.